R/fit.R

```diff
@@ -415,12 +418,21 @@ mmrm <- function(formula,
     drop_visit_levels = control$drop_visit_levels,
     allow_na_response = FALSE
   )
-  fit <- fit_single_optimizer(
-    tmb_data = tmb_data,
-    formula_parts = formula_parts,
-    control = control
-  )
-
+  fit <- list()
+  while ((length(fit) == 0) && length(control$optimizers) > 0) {
+    fit <- fit_single_optimizer(
+      tmb_data = tmb_data,
+      formula_parts = formula_parts,
+      control = control
+    )
+    if (length(fit) == 0) {
+      warning(paste0(
+        "Divergence with optimizer ", names(control$optimizers[1L]), " due to problems: ",
+        toString(attr(fit, "divergence"))
+      ))
+    }
+    control$optimizers <- control$optimizers[-1]
```

Comment:
so this part we modified the `control$optimizers`, and in line 451 we still need the names of all these optmizers to report the error. currently it says

```
Chosen optimizer '' led to problems during model fit:
```